deploy.sh

```diff
@@ -0,0 +1,287 @@
+        distro="jammy"
+        break ;;
+      "noble(Ubuntu 24.04)")
+        distro="noble"
+        break ;;
+    esac
+  done
+
+  if [ "$ARCH" = "x86_64" ]; then
+    ARCH="amd64"
+  fi
+
+  filename="postgresql-$1-pg-search_${LATEST_RELEASE_VERSION}-1PARADEDB-${distro}_${ARCH}.deb"
+  url="https://github.com/paradedb/paradedb/releases/latest/download/${filename}"
+
+  echo "Downloading ${url}"
+
+  curl -L "$url" > "$filename" || false
+
+  sudo apt install ./"$filename" -y || false
+}
+
+# Installs latest RPM package
+installRPM(){
+  filename="pg_search_$1-$LATEST_RELEASE_VERSION-1PARADEDB.el9.${ARCH}.rpm"
+  url="https://github.com/paradedb/paradedb/releases/latest/download/${filename}"
+  echo -e "Installing cURL"
+  sudo dnf install curl || false
+  echo "Successfully Installed cURL✅"
+
+  echo "Downloading ${url}"
+  curl -l "$url" > "$filename" || false
+
+  sudo rpm -i "$filename" || false
+  echo "ParadeDB installed successfully!"
+}
+
+# Installs latest binary for ParadeDB
+installBinary(){
+
+  if [[ "$OSTYPE" = "msys" ]] || [[ "$OSTYPE" = "cygwin" ]]; then
+    echo "Sorry but we do not support windows yet!"
+    echo -e "$EXIT_MSG"
+    exit 1
+  elif [[ "$OSTYPE" = "darwin" ]]; then
+    echo -e "\nOops! We don't have any prebuilt binaries for MacOS right now.\nYou can follow our comprehensive guide to compile ParadeDB from source at: https://github.com/paradedb/paradedb/blob/dev/README.md!"
+    echo -e "$EXIT_MSG"
+    exit 1
+  fi
+
+  # Select postgres version
+  pg_version=
+  echo "Select postgres version(Note: ParadeDB is supported on PG12-16. For other postgres versions, you will need to compile from source.)"
+  versions=("14" "15" "16")
+
+  select vers in "${versions[@]}"
+  do
+    case $vers in
+      "14")
+        pg_version="14"
+        break ;;
+      "15")
+        pg_version="15"
+        break ;;
+      "16")
+        pg_version="16"
+        break ;;
+    esac
+  done
+
+  # Select Base type
+  echo "Select supported file type: "
+  opts=(".deb" ".rpm")
+
+  select opt in "${opts[@]}"
+  do
+    case $opt in
+      ".deb")
+        installDeb $pg_version
+        break ;;
+      ".rpm")
+        installRPM $pg_version
+        break ;;
+    esac
+  done
+}
+
+
+echo -e "=========================================================\n"
+
+echo -e "Hi there!
+
+Welcome to ParadeDB, an open-source alternative to Elasticsearch built on Postgres.\nThis script will guide you through installing ParadeDB.
+
+ParadeDB is available as a Kubernetes Helm chart, a Docker image, and as prebuilt binaries for Debian-based and Red Hat-based Linux distributions.\nHow would you like to install ParadeDB?"
+
+
+echo -e "=========================================================\n"
+
+
+
+
+OPTIONS=("🐳Latest Docker Image" "⬇️ Stable Binary")
+
+
+select opt in "${OPTIONS[@]}"
+do
+  case $opt in
+    "🐳Latest Docker Image")
+      installDocker
+      echo -e "Installation Successfull!\n"
+      break ;;
+    "⬇️ Stable Binary")
```

Comment:
I'm able to go through this flow even though I'm on macOS. We shouldn't let macOS or Windows users install the binaries directly, and instead should just tell them they can build from source or use Docker

Reply:
can you please share the output of echo $OSTYPE ? Ideally it should throw an error message as soon as the binary option is selected

Reply:
```bash
philippemnoel /Users/philippemnoel/Downloads/paradedb/paradedb (phil/extension-docs) $ $OSTYPE
zsh: command not found: darwin23.0
```